### PR TITLE
Script interface maintenance

### LINKED ISSUE #4347: lj-demo.py is broken
The physics in [`samples/lj-demo.py`](https://github.com/espressomd/espresso/blob/cc0b79f46f89a796c689a9b8ea26b5e2e1ee6df7/samples/lj-demo.py) is broken since ESPResSo 4.0. The script is also completely untested, and has diverged, e.g. the NpT option doesn't work because the integrator function signature changed in 4.1. It's possible to fix the sample without a MIDI board.

Tasks:

* Remove the phase diagram plot. It is unclear how it was obtained, and the dot temporarily moves into the solid phase when the volume is increased, which is not correct; as the box volume increases, it takes time for the fluid to occupy the newly available space, which is why the dot temporarily enters the solid region. However a phase diagram describes the state of a system at equilibrium, and during the time it takes for the fluid to occupy the new space, the system is strongly out-of-equilibrium. I also couldn't find an analytical solution for the phase diagram of a simple LJ system, and according to the literature it depends strongly on the cutoff value for the LJ potential (Smit 1992, doi:[10.1063/1.462271](https://aip.scitation.org/doi/10.1063/1.462271)).
* Fix the overflow issue when the GUI sliders are moved too quickly.
* Fix NpT error caused by the change of signature of the NpT integrator.
* Add a minimal test case, if feasible. It's not going to be pretty. `testsuite/scripts/samples/test_visualization_npt.py` can be used as template, should *probably* work with the following patch:

<details>
  <summary>patch (click to unroll)</summary>

```python
def disable_GUI_and_MIDI(code):
    # disable MIDI and traits interface
    breakpoint = 'from traits.api import HasTraits, Any, Range, List, Enum, Float\nfrom traitsui.api import View, Group, Item, CheckListEditor, RangeEditor'
    assert breakpoint in code
    code.replace(
        breakpoint,
        'from unittest.mock import MagicMock()\n' +
        '\n'.join(
            f'{x} = MagicMock()' for y in breakpoint.split('\n') for x in y.split(' import ')[1].split(', ')))
    breakpoint = 'system.thermostat.set_langevin(kT=controls.temperature, gamma=1.0)'
    assert breakpoint in code
    code.replace(breakpoint, 'system.thermostat.set_langevin(kT=1, gamma=1.0)')
    breakpoint = 'new_box = np.ones(3) * controls.volume**(1. / 3.)'
    assert breakpoint in code
    code.replace(breakpoint, 'new_box = [3, 3, 3]')
    breakpoint = 'new_part = controls.number_of_particles'
    assert breakpoint in code
    code.replace(breakpoint, 'new_part = 100')
    # integrate without visualizer
    breakpoint = 't = Thread(target=main_thread)'
    assert breakpoint in code
    code = code.split(breakpoint)[0] + 'controls = Controls()\nmain_thread()'
    return code


sample, skipIfMissingFeatures = importlib_wrapper.configure_and_import(
    "@SAMPLES_DIR@/lj-demo.py", int_n_times=10,
    substitutions=disable_GUI_and_MIDI, cmd_arguments=["--opengl"])
```
</details>

### PR DESCRIPTION
**Pull request by jngrad**
Description of changes:
- API change: rename `espressomd.reaction_ensemble` to `espressomd.reaction_methods`
- feature configuration: remove obsolete `EXPERIMENTAL_FEATURES`
- script interface:
   - use shared pointers instead of `new` expressions in the script interface
   - make type conversion error messages less ambiguous by putting quotes around C++ types
   - require EK node grid indices to be exactly 3 integers (2 indices would skip the exception and return `None`)
- python interface:
   - replace `from X import Y` by `import X` and `X.Y` in samples and python interface
   - use `tqdm`, `numpy` and f-strings more often in the samples
   - remove `lj-demo.py` (fixes  #4347)
   - remove mentions to non-existent functions in the user guide

### CODE REVIEW

File: src/python/espressomd/analyze.pyx [hunk 1]
```diff
@@ -560,7 +559,7 @@ class Analysis:
             raise ValueError("type_list_b has to be a list!")
 
         if r_max is None:
-            box_l = make_array_locked(< Vector3d > box_geo.length())
```
Comment (christophlohrmann) [inline src/python/espressomd/analyze.pyx:563]: How was this never caught by anyone?

File: src/python/espressomd/electrokinetics.pyx [hunk 2]
```diff
@@ -496,13 +491,11 @@ IF ELECTROKINETICS:
...
-                            self.id, self.node[0], self.node[1], self.node[2], value) != 0:
-                        raise Exception("Species has not been added to EK.")
-                else:
-                    raise ValueError(
-                        "Type of property is wrong. Expected: float.")
+                utils.check_type_or_throw_except(
+                    value, 1, float, "Property 'density' has to be a float")
+                if ek_node_set_density(
+                        self.id, self.node[0], self.node[1], self.node[2], value) != 0:
+                    raise Exception("Species has not been added to EK.")
```
Comment (christophlohrmann) [inline src/python/espressomd/electrokinetics.pyx]: That's a very broad Exception

**Review by christophlohrmann — DISMISSED** [review]
PR seems very fine to me. One point though:
As far as I know, we need lj-demo.py for Tag der Wissenschaft. Is there anyone actively working to fix the demo?

File: src/python/espressomd/analyze.pyx (hunk 1 above)
Comment (jngrad) [reply src/python/espressomd/analyze.pyx:563]: I'm not sure, there should have been a warning about the redundant cast according to the [Cython docs on type casting](https://cython.readthedocs.io/en/stable/src/userguide/language_basics.html#type-casting).

**Comment by jngrad** [comment]
> As far as I know, we need lj-demo.py for Tag der Wissenschaft. Is there anyone actively working to fix the demo?

Not that I know. I sent an email to David and Julian on March 22 to inform them of the planned removal and invite them to voice their objection, but got no reply. For the TdW event, people can use PyStar (#4382). When I used `lj-demo.py` for TdW 2019, people asked why the LJ simulation was following `P=VNRT` instead of `PV=NRT`, and I had no answer. The sample is broken since release 4.0.0, and unfortunately no one here was successful in implementing a working version. It is time to retire it.

File: src/python/espressomd/electrokinetics.pyx (hunk 2 above)
Comment (jngrad) [reply src/python/espressomd/electrokinetics.pyx]: done

**Review by reinaual — APPROVED** [review]

**Pull request merged**